cpp/src/groupby/hash/groupby.cu

```diff
@@ -118,74 +120,135 @@ bool constexpr is_hash_aggregation(aggregation::Kind t)
   return array_contains(hash_aggregations, t);
 }
 
+bool is_hash_aggregation(aggregation const& agg)
+{
+  if (agg.kind == aggregation::NTH_ELEMENT) {
+    auto const& n = dynamic_cast<cudf::detail::nth_element_aggregation const&>(agg)._n;
```

Comment:
I thought that too. The issue is that `aggregation` is a `virtual` base class for `nth_element_aggregation`. I found no option other than to use `dynamic_cast`.

Reply:
For reference: https://godbolt.org/z/K71Ea94qh:

```
<source>:9:65: error: cannot convert from pointer to base class 'aggregation' to pointer to derived class 'sum_aggregation' because the base is virtual
    9 | sum_aggregation* p_sum_agg = static_cast<sum_aggregation*>(p_agg);
      |                                                                 ^
Compiler returned: 1
```
The offset to be applied to the `aggregation` base pointer to the `sum_aggregation` sub-object is not known at compile time, because the base class is virtual. :]